app/controllers/api_controller/provision_requests.rb

```diff
@@ -18,5 +18,25 @@ def create_resource_provision_requests(type, _id, data)
       MiqProvisionVirtWorkflow.from_ws(version_str, @auth_user_obj, template_fields, vm_fields, requester, tags,
                                        additional_values, ems_custom_attrs, miq_custom_attrs)
     end
+
+    def deny_resource_provision_requests(type, id, data)
+      api_action(type, id) do |klass|
```

Comment:
updated this, according to `controllers/api_controller/vms.rb`, hope it is better now